juju-terraform-provider-k8s-controller-full-bundle/providers.tf

```diff
@@ -0,0 +1,21 @@
+# This ones will be downloaded on `terraform init`
+terraform {
+  required_providers {
+    juju = {
+      version = "~> 0.4.3"
+      source  = "juju/juju"
+    }
+  }
+}
+
+
+# Provider configuration section. 
+# See: https://registry.terraform.io/providers/juju/juju/latest/docs#schema
+provider "juju" {
+    # This block could be left empty thus making the provider make 
+    # a subcall to the `juju` CLI to get this data by itself.
+    controller_addresses = var.k8s_juju_controller_address
+    username = var.k8s_juju_controller_username
+    password = var.k8s_juju_controller_password
+    ca_certificate = file(var.k8s_juju_controller_ca_certificate_path)
```

Comment:
I don't love this, how would this look when using the `juju` CLI?

Reply:
```
provider "juju" {}
```

Reply:
Just a minor comment here.

As mentioned running `provider "juju" {}` extracts the connection details using the Juju CLI (see [here](https://github.com/juju/terraform-provider-juju/blob/main/internal/juju/utils.go#L68)). I would go for the solution explored here to not depend on the local CLI.

Reply:
I think I'm somewhere in between! I wouldn't parameterise these as variables in this file (or variables.tf). I'd do as @moisesbenzan suggested, and if people need to override any of these values they can do so [using environment variables](https://registry.terraform.io/providers/juju/juju/latest/docs#optional) which will override this at runtime if needed.

Reply:
My only concerned with using the environment variables approach is that it can be removed from the provider in a future as it is still under heavy development +  it makes using this kubeflow model as a part of a greater TF model harder as it will not have any direct inputs. This is me looking into the future for when we'd want to do more robust deployments such as deploy K8s with TF and then deploy this on top in one automated step.

Reply:
I don't think we'll remove the environment variables - that would make our provider "weird" next to the best of breed (AWS, Azure, etc.). I think you can take that feature to the bank :)

Reply:
Applied suggested changes.